### ISSUE TITLE
Blowers Masel Arrhenius trained and evaluated with T-varying ∆Hr(T) ?

### ISSUE THREAD

**Issue by rwest** [ReactionMechanismGenerator/RMG-Py#1748]
### Bug Description

Here
https://github.com/ReactionMechanismGenerator/RMG-Py/blob/3e37656f71980784e9a0791c1c3e14273480ade3/rmgpy/data/kinetics/family.py#L3502
an `ArrheniusBM` is converted to an `Arrhenius` using the ∆Hrxn evaluated at the temperature of interest, something like 1000K, rather than at 298K.

Looking at the code used to fit the BM expressions here 
https://github.com/ReactionMechanismGenerator/RMG-Py/blob/91cdd1da3c5a7d88b1c53a82f683ff7ddbe2e871/rmgpy/kinetics/arrhenius.pyx#L616
perhaps this is correct, or at least consistent (we shouldn't change one without changing both.)

Skimming the Blowers and Masel paper https://doi.org/10.1002/aic.690461015 I can't see it explicitly say what temperature it's evaluated at (it just says "∆Hr is the heat of reaction", and later "[we have] looked up the heats  of reaction [for the 151 reactions in Table 1]") but given that it doesn't talk about barrier heights changing at different temperatures, just barrier heights, it doesn't make sense to me for ∆Hr to be a function of temperature. If Ea is f(∆Hr) and Ea independent of T then ∆Hr is independent of T.   I would guess they looked up the 298K value, but as they don't actually give any of the values in the paper I'm not sure how to check. If we're re-training our own parameters then probably its up to us anyway - as long as we're consistent.

To my knowledge (and ability to search) all other calls of `to_arrhenius` in RMG, whether called on instances of ArrheniusEP, SurfaceArrheniusBEP, StickingCoefficientBEP, or ArrheniusBM, all use ∆Hr298.

My gut says ∆Hr shouldn't be a function of Temperature, and so  the BM fitting and evaluating segments linked above are wrong.  But I've run out of time to dig deeper.

**Comment by mjohnson541** [comment]
My understanding is that ∆Hr is a relatively weak function of temperature.  When fitting we evaluate ∆Hr(T) to be thorough in fitting, but I don't think we lose much accuracy when we convert to Arrhenius, usually at T=1000 K (because that's what the trees are currently fitted at).

**Comment by rwest** [comment]
As we set about implementing Blowers Masel rate expressions in Cantera (https://github.com/Cantera/cantera/pull/987 ) we have come across this issue again. Should the Blowers Masel expression be evaluated using  ∆Hr298 or ∆Hr(T) ?

Yes, Matt's right ∆Hrxn is a _pretty_ weak function of T, but when @12Chao converts all H abstractions (except those with negative barriers) into Blowers Masel expressions, fitted at 298K but evaluated at T, the ignition delay of n-heptane changes about 3%. So not very big, but not zero.

I guess we have a few questions. 1) what did Blowers and Masel intend? 2) what actually works best? 3) what will users expect? (we have chance to influence this, as for now we are the only users).

**Comment by mjohnson541** [comment]
For (1) the paper (https://aiche.onlinelibrary.wiley.com/doi/abs/10.1002/aic.690461015) almost comically doesn't even have the word temperature in the whole document. In general it talks about dHrxn like it isn't temperature dependent. As far as what they're actually using they say: "In order to do so we have taken all of the hydrogen transfer reactions listed in the NIST tables Westley 1980., and looked up the heats of reaction." Which makes me think they're using dHrxn298 or dHrxn0.

**Comment by mjohnson541** [comment]
One thing I particularly don't like about evaluating the BM using dHrxn(T) is that it causes programmatic branching in the rate evaluation which makes it more expensive to differentiate in the reverse-mode and much harder to generate symbolic representations for. Of course NASA polynomials have the same issue, but in an ideal world I'd like to move away from them as well.

**Comment by mjohnson541** [comment]
So I think we might want to think about how we format things this way: 

RMG est format  => RMG output => Simulator format

We use RMG to generate the rate/thermochemistry estimates in some programmatic format, we write the RMG mechanism to a file in some format and when the simulator reads the file it compiles it into a third format. 

In the first case we're generating the estimates so the format is whatever our estimators spit out and we really only care that the format allows the estimators to perform well. In the simulator format we care that we've represented the estimation format properly, but we also care a lot about our ability to evaluate it optimally. The RMG output can be anything as long as it has enough information to generate simulator format. 

Following the solvation example. We could have RMG output that is solvent specific. We could also output something much more similar to the RMG est format and let the simulator interpret that to whatever format it thinks is most convenient. These put different burdens on different software, but I think neither of these prevents the simulator from compiling its own kinetics format at the end.

**Comment by rwest** [comment]
@sevyharris and I just spent a while getting very confused as to why our Blowers-Masel fit was *not* fitting the training reaction when evaluated with H(T) but only with H(1000K). Turns out it's because we had a single training reaction, and this line 
https://github.com/ReactionMechanismGenerator/RMG-Py/blob/336273d2be60d3c6fb030c70b346ec97d642d8e0/rmgpy/kinetics/arrhenius.pyx#L601
added in https://github.com/ReactionMechanismGenerator/RMG-Py/pull/1978
shows that a single training reaction is fitted using H(1000K) not H(T).

Having been thinking about this quite a bit (in the context of adding BM to Cantera) I'm leaning towards the conclusion that Blowers and Masel intended that it should be H(298K) not H(T).   In any case, I think we should be consistent!
